packages/rich-text/src/normalise-formats.js

```diff
@@ -18,22 +12,28 @@ import { isFormatEqual } from './is-format-equal';
  * @return {Object} New value with normalised formats.
  */
 export function normaliseFormats( value ) {
-	const refs = [];
-	const newFormats = value.formats.map( ( formatsAtIndex ) =>
-		formatsAtIndex.map( ( format ) => {
-			const equalRef = find( refs, ( ref ) =>
-				isFormatEqual( ref, format )
-			);
+	const newFormats = value.formats.slice();
+
+	newFormats.forEach( ( formatsAtIndex, index ) => {
```

Comment:
Thanks, I'll have a look at this

Reply:
Hm, the reducer is possible, but it makes it equally complex... Plus you'd have to start with an array that has the same length and fill it in. The array should stay sparse. I don't really mind either way.

Reply:
> Hm, the reducer is possible, but it makes it equally complex... Plus you'd have to start with an array that has the same length and fill it in. The array should stay sparse. I don't really mind either way.

It's definitely not a blocking issue, and if it can't help simplify, would be fine to skip.